src/test/java/seedu/address/ui/testutil/GuiTestAssert.java

```diff
@@ -34,8 +37,60 @@ public static void assertCardDisplaysPerson(Person expectedPerson, PersonCardHan
         assertEquals(expectedPerson.getPhone().value, actualCard.getPhone());
         assertEquals(expectedPerson.getEmail().value, actualCard.getEmail());
         assertEquals(expectedPerson.getAddress().value, actualCard.getAddress());
-        assertEquals(expectedPerson.getTags().stream().map(tag -> tag.tagName).collect(Collectors.toList()),
-                actualCard.getTags());
+        assertTagEquals(expectedPerson, actualCard);
+    }
+
+    /**
+     * Asserts that{@code actualCard} displays the same tag and colour of {@code expectedPerson}
+     * @param expectedPerson
+     * @param actualCard
+     */
+    private static void assertTagEquals(Person expectedPerson, PersonCardHandle actualCard) {
+        List<String> expectedTags = expectedPerson.getTags()
+                .stream().map(tag -> tag.tagName)
+                .collect(Collectors.toList());
+
+        assertEquals(expectedTags, actualCard.getTags());
+
+        expectedTags.forEach(tag -> assertEquals(Arrays
+                .asList(LABEL_STYLE, getTagColorStyleFor(tag)), actualCard.getTagStyleClasses(tag)));
+    }
+
+    /**
+     *To be updated
```

Comment:
Might want to set it as a `TODO:` so that it will be highlighted in your IDE. Make sure the basic functionalities of the program is working before you push it to master, as this is the "gold standard" of your application

Reply:
updated